### ISSUE TITLE
Does matrix A need to be diagonal?

### ISSUE THREAD

**Issue by houghtonweihu** [johnma2006/mamba-minimal#14]
In your file: https://github.com/johnma2006/mamba-minimal/blob/master/model.py, you have:

   A = repeat(torch.arange(1, args.d_state + 1), 'n -> d n', d=args.d_inner)
   self.A_log = nn.Parameter(torch.log(A))
   A = -torch.exp(self.A_log.float())  # shape (d_in, n)

Does matrix A need to be diagonal?

   Thanks!

**Comment by lukaemon** [comment]
> I got an answer from Dr Dao: d_in is actually the batch dim.

What do you mean `d_in` is batch dim? 
In Dao's original code: https://github.com/state-spaces/mamba/blob/main/mamba_ssm/modules/mamba_simple.py#L104
```python
self.d_inner = int(self.expand * self.d_model)

A = repeat(
    torch.arange(1, self.d_state + 1, dtype=torch.float32, device=device),
    "n -> d n",
    d=self.d_inner,
).contiguous()        
```
How could it be batch dim?

**Comment by Joeland4** [comment]
> I got an answer from Dr Dao: d_in is actually the batch dim.

what is the batch dim？
